### ISSUE TITLE
Unable to expand into multiple dynamic dims (Aten.View Collapse/Expand)

### ISSUE THREAD

**Issue by JakopinA** [llvm/torch-mlir#1390]
Hi, I've been working on a solution for Aten.view collapse/expand for #1353. With the following test, I've been trying to implement support for dynamic collapse and expansion:

`class ViewFlattenAndExpandModule(torch.nn.Module):
    def __init__(self):
        super().__init__()

    @export
    @annotate_args([
        None,
        ([-1, -1], torch.float32, True),
    ])

    def forward(self, a):
        return a.view(a.size(1), a.size(0))

@register_test_case(module_factory=lambda: ViewFlattenAndExpandModule())
def ViewFlattenAndExpandModule_basic(module, tu: TestUtils):
    module.forward(tu.rand(64,128))`

This is the error I've been getting:

https://gist.github.com/JakopinA/f23408454b10c49945defd01086aa1bd

@silvasean

**Comment by qedawkins** [comment]
The problem here is that tensor.collapse and tensor.expand ops don't allow for the dynamic flatten then expand so the fallback "flatten then expand" approach we wanted are contingent on changes in MLIR for dynamic cases other than [?, ?] -> [at most one dynamic dim].

**Comment by silvasean** [comment]
Was this fixed by #1353 ?

**Comment by qedawkins** [comment]
No, this is an MLIR issue with the tensor.expand_shape op, so we can probably move it to llvm-project.
